src/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -156,11 +156,19 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
 
   private newCaseField(id: string, item, index, isNew = false) {
     const isNotAuthorisedToUpdate = !isNew && this.isNotAuthorisedToUpdate(index);
+
+    const fieldType = plainToClassFromExist(new FieldType(), this.caseField.field_type.collection_field_type);
+    if (fieldType.complex_fields) {
+      fieldType.complex_fields
+        .filter((cf: CaseField) => cf.show_condition !== null)
```

Comment:
Do we know for sure this is always `null` when it's not set? It's never `undefined` or `''`? `!!cf.show_condition` would account for all of those, which might be a better option.